prismjs/prismjs-module-tests.ts

```diff
@@ -0,0 +1,26 @@
+import * as Prism from "prismjs";
+
+var element = document.createElement("code");
+var callback = (element: Element) => console.log(element);
+
+Prism.highlightElement(element, false, callback);
+Prism.highlightElement(element, false);
+Prism.highlightAll(true, callback);
+Prism.highlightAll(true);
+
+const hookCallback: Prism.HookCallback = env => null;
+Prism.hooks.add("before-highlightall", hookCallback);
+Prism.hooks.add("future-hook", hookCallback);
+
+var language = "js";
+var tokens = Prism.tokenize("var n = 1;", Prism.languages[language]);
+(function visit(token: Prism.TokenNode): Prism.TokenNode {
+    if (typeof token === "string") {
+        return token;
+    } else if (Array.isArray(token)) {
+        return token.map(visit) as Prism.TokenNode;
```

Comment:
I need the help of someone with more "type" experience than I to figure out what the correct thing to do here is so that this function doesn't have to cast the return result.